app/service/data_svc.py

```diff
@@ -31,19 +43,61 @@ def __init__(self):
                            schedules=[], plugins=[], obfuscators=[], objectives=[])
         self.ram = copy.deepcopy(self.schema)
 
+    @staticmethod
+    def _iter_data_files():
+        """Yield paths to data files managed by caldera.
+
+        The files paths are relative to the root caldera folder, so they
+        will begin with "data/".
+
+        Note:
+            This will skip any files starting with '.' (e.g., '.gitkeep').
+        """
+        for data_glob in DATA_FILE_GLOBS:
+            for f in glob.glob(data_glob):
+                yield f
+
+    @staticmethod
+    def _delete_directory_contents(path):
+        """Delete all files and subdirectories under `path`.
+
+        Note:
+            This uses `glob` and thus, ignores top-level files
+            that start with a '.' (e.g., '.gitkeep'. Any subdirectories
+            are deleted entirely (even if they contain '.' files).
+        """
+        if not os.path.exists(path):
+            return
+
+        if not os.path.isdir(path):
+            raise ValueError(f'Input path must be a directory. Received {path}')
+
+        for path in glob.glob(f'{path}/*'):
+            if os.path.isdir(path):
+                shutil.rmtree(path)
+            else:
+                os.remove(path)
+
     @staticmethod
     async def destroy():
-        if os.path.exists('data/object_store'):
-            os.remove('data/object_store')
+        """Clear the caldera data directory and server state.
 
-        for d in ['data/results', 'data/adversaries', 'data/abilities', 'data/facts', 'data/sources', 'data/payloads', 'data/objectives']:
-            for f in glob.glob('%s/*' % d):
-                if f.startswith('.'):  # e.g., .gitkeep
-                    continue
-                elif os.path.isdir(f):
-                    shutil.rmtree(f)
-                else:
-                    os.remove(f)
+        This moves all data files and the object store to the data backup directory.
+        The original data file paths are preserved under the backup directory.
+
+        Example (original path -> new backup path):
+            data/results/23deddf-ff3f2.yml -> backup/data/results/23deddf-ff3f2.yml
+        """
+        if os.path.exists(DATA_BACKUP_DIR):
+            DataService._delete_directory_contents(DATA_BACKUP_DIR)
+        else:
+            os.mkdir(DATA_BACKUP_DIR)
+
+        for file_path in DataService._iter_data_files():
+            src_dir, src_filename = os.path.split(file_path)
+            dst_dir = os.path.join(DATA_BACKUP_DIR, src_dir)
+            os.makedirs(dst_dir, exist_ok=True)
+            shutil.move(file_path, os.path.join(dst_dir, src_filename))
```

Comment:
would it be easy here to compress with a timestamp in the name?

Reply:
Yep, pretty simple. Do you think the backup directory should be cleared out every `--fresh` run, or should we allow multiple timestamped backup tarballs in there?

Reply:
I changed the behavior to not clear out the backup directory and instead just always create a new, timestamped tarball